src/worker.js

```diff
@@ -54,16 +39,19 @@ create().onRequest('job', ({ contents, type, config, filePath, projectPath, rule
   const configPath = Helpers.getConfigPath(fileDir)
   const relativeFilePath = Helpers.getRelativePath(fileDir, filePath, config)
 
-  const argv = Helpers.getArgv(type, config, rules, relativeFilePath, fileDir, configPath)
+  const cliEngineOptions =
+    Helpers.getCLIEngineOptions(type, config, rules, relativeFilePath, fileDir, configPath)
 
-  if (type === 'lint') {
-    job.response = lintJob(argv, contents, eslint, configPath, config)
+  const noProjectConfig = (configPath === null || isConfigAtHomeRoot(configPath))
+  if (noProjectConfig && config.disableWhenNoEslintConfig) {
+    job.response = []
+  } else if (type === 'lint') {
+    const report = lintJob({ cliEngineOptions, contents, eslint, filePath })
+    job.response = report.results.length ? report.results[0].messages : []
   } else if (type === 'fix') {
-    job.response = fixJob(argv, eslint)
+    job.response = fixJob({ cliEngineOptions, contents, eslint, filePath })
```

Comment:
I think we should explore using `executeOnText` for fixes, but in a separate PR. I'd like to keep this PR small in scope so we're not changing too much at once.